src/plone/restapi/services/upgrade/post.py

```diff
@@ -0,0 +1,28 @@
+from plone.restapi.services import Service
+from plone.restapi.deserializer import json_body
+from Products.CMFCore.utils import getToolByName
+from Products.CMFPlone.browser.admin import Upgrade
+from zope.interface import alsoProvides
+
+import plone
+
+
+class UpgradeSitePost(Service):
+    def reply(self):
+        # Disable CSRF protection
+        if "IDisableCSRFProtection" in dir(plone.protect.interfaces):
+            alsoProvides(self.request, plone.protect.interfaces.IDisableCSRFProtection)
+        data = json_body(self.request)
+        dry_run = data.get("dry_run", False)
+        pm = getToolByName(self.context, "portal_migration")
+        report = pm.upgrade(REQUEST=self.request, dry_run=dry_run)
+        view = Upgrade(self.context, self.request)
+        versions = view.versions()
+        return {
+            "@id": f"{self.context.absolute_url()}/@upgrade",
+            "report": report,
+            "versions": {
+                "instance": versions["instance"],
+                "fs": versions["fs"],
+            },
+        }
```

Comment:
@ericof Should we return a clear status for whether the upgrade was successful? https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/browser/templates/plone-upgrade.pt appears to check this based on whether the versions ended up equal or not.